### PR TITLE
auth: somewhat cleanup global symbols

### PR DESCRIPTION
**Pull request by zeha**
### Short description
- Make many things in auth-main.cc static
- Rename some of the single/two-letter globals to longer names, but mostly those local to auth-main.cc
- Move SOAData, DNSBackend to auth-only include files

I've deliberately ignored the symbols used across many files (QC, PC, ...), in the hope this could still go into 4.7.0.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/auth-main.cc [hunk 1]
```diff
@@ -117,24 +117,25 @@ AuthPacketCache PC; //!< This is the main PacketCache, shared across all threads
 AuthQueryCache QC;
 AuthZoneCache g_zoneCache;
 std::unique_ptr<DNSProxy> DP{nullptr};
-std::unique_ptr<DynListener> dl{nullptr};
+static std::unique_ptr<DynListener> g_DynListener{nullptr};
```
Comment (rgacogne) [inline pdns/auth-main.cc:120]: Just a nit but now that the variable is local to this file, the `g_` prefix might no longer be appropriate :) `s_DynListener` ?
Same for `g_TcpNameserver`, `g_udpNameserver` and `g_udpReceivers`.

File: pdns/auth-main.hh [hunk 2]
```diff
@@ -35,22 +35,13 @@
 #include "tcpreceiver.hh"
 #include "dnsseckeeper.hh"
 
+extern time_t s_starttime;
```
Comment (rgacogne) [inline pdns/auth-main.hh:38]:
```suggestion
extern time_t g_starttime;
```

**Review by rgacogne — APPROVED** [review]
Looks good, a couple nits about the prefixes but nothing that really matters.

File: pdns/auth-main.cc (hunk 1 above)
Comment (zeha) [reply pdns/auth-main.cc:120]: Right, wasn't sure if they should be `s_` then. I'll change that, and maybe move some into mainthread() and out of the global scope.

File: pdns/auth-main.hh (hunk 2 above)
Comment (zeha) [reply pdns/auth-main.hh]: done, thanks!

File: pdns/auth-main.cc (hunk 1 above)
Comment (zeha) [reply pdns/auth-main.cc]: all done.

File: pdns/auth-main.cc [hunk 3]
```diff
@@ -127,14 +127,15 @@ static vector<std::shared_ptr<UDPNameserver>> g_udpReceivers;
 NetmaskGroup g_proxyProtocolACL;
 size_t g_proxyProtocolMaximumSize;
 
-void mainthread();
+// auth-carbon.cc
+void carbonDumpThread();
```
Comment (Habbie) [inline pdns/auth-main.cc:131]: this avoids having `auth-carbon.hh`? maybe add a comment explaining that

**Review by Habbie — COMMENTED** [review]
one small question, otherwise approved. Happy to get this into 4.7 too.

File: pdns/auth-main.cc (hunk 3 above)
Comment (zeha) [reply pdns/auth-main.cc:134]: yes. added a comment

**Comment by zeha** [comment]
Rebased to fix conflict.

**Pull request merged**